pxemgr/pxemanager.go

```diff
@@ -118,19 +121,49 @@ func PXEManager(c PXEManagerConfiguration, cluster *hostmgr.Cluster) (*pxeManage
 		mu: new(sync.Mutex),
 	}
 
+	// check for deprecated EtcdDiscoveryUrl
+	if mgr.cluster.Config.EtcdDiscoveryURL != "" && mgr.cluster.Config.DefaultEtcdClusterToken == "" {
+		// transform discovery url to token
+		parts := strings.Split(mgr.cluster.Config.EtcdDiscoveryURL, "/")
+		token := parts[len(parts)-1]
+		baseUrl := strings.Join(parts[:len(parts)-1], "/")
+
+		if mgr.useInternalEtcdDiscovery {
+			// convert token to internal etcd discovery
+			err := mgr.cluster.StoreEtcdDiscoveryToken(mgr.etcdEndpoint, token, mgr.defaultEtcdQuorumSize)
+			if err != nil {
+				glog.Fatalf("Can't store discovery token in etcd.", baseUrl, mgr.etcdDiscoveryUrl)
+			}
+
+			glog.Warningf("Transferred etcd token to internal discovery. Note that your machines still have the old discovery url in their cloud-config and that you need to transfer the current member data yourself.")
+		} else if mgr.etcdDiscoveryUrl != baseUrl {
+			glog.Fatalf("Deprecated EtcdDiscoveryURL ('%s') in your cluster.json differs from the --etcd-discovery parameter ('%s').", baseUrl, mgr.etcdDiscoveryUrl)
+		}
+		mgr.cluster.Config.EtcdDiscoveryURL = ""
```

Comment:
Shouldn't we  set `mgr.cluster.Config.EtcdDiscoveryURL = mgr.etcdDiscoveryUrl` ? I mean to the internal discovery url